pil-analyzer/src/evaluator.rs

```diff
@@ -599,7 +599,7 @@ enum Operation<'a, T> {
 struct Evaluator<'a, 'b, T: FieldElement, S: SymbolLookup<'a, T>> {
     symbols: &'b mut S,
     local_vars: Vec<Arc<Value<'a, T>>>,
-    type_args: HashMap<String, Type>,
+    type_args: Arc<HashMap<String, Type>>,
```

Comment:
I don't see where this `Arc` is useful - am I missing something?

Reply:
https://github.com/powdr-labs/powdr/blob/a7577de0fd38c58f9c25059c23b955e04d3191a5/pil-analyzer/src/evaluator.rs#L982

<img width="1660" alt="Screenshot 2024-08-05 at 11 08 09" src="https://github.com/user-attachments/assets/30712b73-2ec6-4e91-96f5-77d1ccb58770">

Reply:
Ok, makes sense! :)